device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -1216,4 +1221,101 @@ private void verifyMethodsAreSupported()
+            telemetryMessage.setContentEncoding(PayloadConvention.getPayloadEncoder().getContentEncoding().name());
+            telemetryMessage.setContentTypeFinal(PayloadConvention.getPayloadSerializer().getContentType());
+        }
+
+        sendEventAsync(telemetryMessage, callback, callbackContext);
+    }
+
+    /**
+     * Set the global command callback handler.
+     * @param callback The callback to be used for commands.
+     * @param callbackContext An optional user context to be sent to the callback.
+     */
+    public void subscribeToCommands(DeviceMethodCallback callback, Object callbackContext)
+    {
+        // Subscribe to methods default handler internally and use the callback received internally to invoke the user supplied command callback.
+        // TODO Implement command handler
+    }
+
+
+    /**
+     * Retreieve the client properties.
+     * @param callback The callback to be used for receiving client properties.
+     * @param callbackContext An optional user context to be sent to the callback.
+     */
+    public void getClientProperties(ClientPropertiesCallback callback, Object callbackContext)
+    {
+        // In Java we don't return an object for the DeviceTwin, but instead we pass a Map<String, Object> in the form of a Device
+        // this map gets populated with a number of callbacks.
+        try
+        {
+            twin.getClientProperties(callback, callbackContext);
+        }
+        catch (Throwable ex)
+        {
+            throw ex;
+        }
+    }
+
+    /**
+     * Update the client properties.
+     * @param clientProperties The client properties to send.
+     * @param callback The callback to be used for receiving client properties.
+     * @param callbackContext An optional user context to be sent to the callback.
+     */
+    public void updateClientProperties(ClientPropertyCollection clientProperties, IotHubEventCallback callback, Object callbackContext) throws IOException
+    {
+        if (clientProperties == null)
+        {
+            throw new IllegalArgumentException("clientProperties property cannot be null");
+        }
+
+        clientProperties.Convention = PayloadConvention;
+
+        verifyRegisteredIfMultiplexing();
+        verifyTwinOperationsAreSupported();
+
+        verifyReportedProperties(clientProperties.getCollectionAsSetOfProperty());
+
+        this.twin.updateClientProperties(clientProperties, null, null,null, callback, callbackContext);
+    }
+
+    /**
+     * Set the global writable properties callback handler.
+     * @param writablePropertyUpdateCallback The callback to be used for writable properties.
+     * @param callbackContext An optional user context to be sent to the callback.
+     * @throws IOException
+     */
+    public void subscribeToWritablePropertiesEvent(WritablePropertiesRequestsCallback writablePropertyUpdateCallback, Object callbackContext) throws IOException
```

Comment:
```suggestion
    public void subscribeToWritablePropertiesEvents(WritablePropertiesRequestsCallback writablePropertyUpdateCallback, Object callbackContext) throws IOException
```

Since you expect the event to happen more than once?